providers/azure.go

```diff
@@ -177,21 +177,27 @@ func (p *AzureProvider) Redeem(ctx context.Context, redirectURL, code string) (*
 	return session, nil
 }
 
-// EnrichSession finds the email to enrich the session state
+// EnrichSession finds the email and groups claim to enrich the session state
 func (p *AzureProvider) EnrichSession(ctx context.Context, s *sessions.SessionState) error {
-	if s.Email != "" {
-		return nil
+	if s.Email == "" {
+		// add email
+		email, err := p.getEmailFromProfileAPI(ctx, s.AccessToken)
+		if err != nil {
+			return fmt.Errorf("unable to get email address: %v", err)
+		}
+		if email == "" {
+			return errors.New("unable to get email address")
+		}
+		s.Email = email
 	}
 
-	email, err := p.getEmailFromProfileAPI(ctx, s.AccessToken)
-	if err != nil {
-		return fmt.Errorf("unable to get email address: %v", err)
-	}
-	if email == "" {
-		return errors.New("unable to get email address")
+	if s.Groups == nil {
```

Comment:
If groups is a list, do we need to check for non nil here or do we just want to check for the length of the list?